--- python-memcached-1.43.orig/memcache.py	2012-02-10 15:50:59.306985568 -0500
+++ python-memcached-1.43.orig/memcache.py	2012-02-10 15:53:55.896317256 -0500
@@ -160,7 +160,7 @@ 
         self.unpickler = unpickler
         self.persistent_load = pload
         self.persistent_id = pid
-        
+
     def set_servers(self, servers):
         """
         Set the pool of servers used by this client.
@@ -185,7 +185,10 @@ 
         data = []
         for s in self.servers:
             if not s.connect(): continue
-            name = '%s:%s (%s)' % ( s.ip, s.port, s.weight )
+            if s.family == socket.AF_INET:
+                name = '%s:%s (%s)' % ( s.ip, s.port, s.weight )
+            else:
+                name = 'unix:%s (%s)' % ( s.address, s.weight )
             s.send_cmd('stats')
             serverData = {}
             data.append(( name, serverData ))
